### PR TITLE
Upgrade to 0.9.16

### PR DESCRIPTION
**Pull request by rhuttman**

### CODE REVIEW

File: runtime/neumann/src/lib.rs [hunk 1]
```diff
@@ -317,6 +318,7 @@ impl frame_system::Config for Runtime {
 	type SS58Prefix = SS58Prefix;
 	/// The action to take on a Runtime Upgrade
 	type OnSetCode = cumulus_pallet_parachain_system::ParachainSetCode<Self>;
+	type MaxConsumers = frame_support::traits::ConstU32<16>;
```
Comment (irsal) [inline runtime/neumann/src/lib.rs:321]: what does this do?

**Review by irsal — APPROVED** [review]
💯

File: runtime/neumann/src/lib.rs [hunk 2]
```diff
@@ -188,6 +188,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	impl_version: 0,
 	apis: RUNTIME_API_VERSIONS,
 	transaction_version: 2,
+	state_version: 0,
```
[the remark this answers is not in the document]
Comment (rhuttman) [reply runtime/neumann/src/lib.rs:191]: /// Version of the state implementation used by this runtime.
/// Use of an incorrect version is consensus breaking.

I still need to do a deep dive on when each version needs to change

File: runtime/neumann/src/lib.rs (hunk 1 above)
Comment (rhuttman) [reply runtime/neumann/src/lib.rs:321]: /// The maximum number of consumers allowed on a single account.

where the number of consumers on an account:

/// The number of other modules that currently depend on this account's existence. The account
/// cannot be reaped until this is zero.

**Review by justinzhou93 — COMMENTED** [review]
couple of noob questions, bear with me. also, do we need to bump spec version as well here?

File: node/src/command.rs [hunk 3]
```diff
@@ -333,11 +336,24 @@ impl CliConfiguration<Self> for RelayChainCli {
...
+		self.base.base.prometheus_config(default_listen_port, chain_spec)
 	}
 
-	fn init<C: SubstrateCli>(&self) -> Result<()> {
+	fn init<F>(
+		&self,
+		_support_url: &String,
+		_impl_version: &String,
+		_logger_hook: F,
+		_config: &sc_service::Configuration,
```
[the remark this answers is not in the document]
Comment (rhuttman) [reply node/src/command.rs:352]: required

**Review by justinzhou93 — APPROVED** [review]

File: node/src/command.rs [hunk 4]
```diff
@@ -256,8 +258,9 @@ pub fn run() -> Result<()> {
 				let parachain_account =
 					AccountIdConversion::<polkadot_primitives::v0::AccountId>::into_account(&id);
 
-				let block: Block =
-					generate_genesis_block(&config.chain_spec).map_err(|e| format!("{:?}", e))?;
+				let state_version = Cli::native_runtime_version(&config.chain_spec).state_version();
```
[the remark this answers is not in the document]
Comment (irsal) [reply node/src/command.rs:261]: let's bump it right before release, no need to do it in this PR

[the remark this answers is not in the document]
Comment (irsal) [reply node/src/command.rs:261]: it's cleaner that way...also anytime we upgrade the client, we probably have to bump node/Cargo.toml

**Review by justinzhou93 — APPROVED** [review]
re

**Pull request merged**